homeassistant/components/avocent_dpdu/config_flow.py

```diff
@@ -0,0 +1,91 @@
+"""Config flow for Avocent Direct PDU integration."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from avocentdpdu.avocentdpdu import AvocentDPDU
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_COUNT, CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    DEFAULT_NUM_OUTLETS,
+    DEFAULT_PASSWORD,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    LOGGER,
+)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): str,
+        vol.Required(CONF_PASSWORD, default=DEFAULT_PASSWORD): str,
+        vol.Required(CONF_COUNT, default=DEFAULT_NUM_OUTLETS): vol.In([8, 16]),
```

Comment:
Could this number be automatically discovered from the PDU (eg if 8 and 16 are the only choices,  try to query outlet 16 and see if it works or not).

Reply:
Sure, now it polls the status and instantiates that number of outlets.